### PR TITLE
Add rebuilderd role

### PR DESCRIPTION
**Pull request by kpcyrd**
- [x] deploy a secret so `rebuilderd-sync` can authenticate with rebuilderd
- [x] deploy a worker
- [ ] configure rebuilderd in nginx proxy

### CODE REVIEW

File: ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#real_ip_header = "X-Real-IP"
+## Set a default endpoint for rebuildctl. This is especially useful for the sync timer.
+#endpoint = "http://127.0.0.1:8484"
+
+## A random cookie for administration is generated at startup and written to /var/lib/rebuilderd/auth-cookie
+## You can set this to a fixed value here. Use `pwgen -1s 32` to generate one.
+## rebuildctl is searching for this cookie in ~/.config/rebuilderd.conf, /etc/rebuilderd.conf and
+## /var/lib/rebuilderd/auth-cookie in that order.
+#[auth]
+#cookie = "INSECURE"
```
Comment (SantiagoTorres) [inline ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf]: I think other than this being templatable this PR looks good right?

**Review by SantiagoTorres — COMMENTED** [review]
overall, shall we move the cookie bit to a different PR and go ahead? what do you think @hojoung97 ?

File: ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf (hunk 1 above)
Comment (kpcyrd) [reply ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf]: That'd work. There's currently no vault yet, so if I'd go ahead and create one everybody would need to set it up as well or they won't be able to deploy anymore.

File: ansible/roles/purdue-rebuilderd/files/rebuilderd.service [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[Unix]
```
Comment (hojoung97) [inline ansible/roles/purdue-rebuilderd/files/rebuilderd.service:1]: I think this is a small typo

Comment (hojoungjang) [inline ansible/roles/purdue-rebuilderd/files/rebuilderd.service]: I think this is a small typo

**Review by hojoungjang — COMMENTED** [review]
@SantiagoTorres @kpcyrd  Hello, thanks for looping me in. I had a chance to look. I think this looks great and honestly, I am mostly learning from you guys :)

File: ansible/roles/purdue-rebuilderd/files/rebuilderd.service (hunk 2 above)
Comment (kpcyrd) [reply ansible/roles/purdue-rebuilderd/files/rebuilderd.service]: Good catch, thanks!

File: ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf (hunk 1 above)
Comment (SantiagoTorres) [reply ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf]: yeah, I think that's ok. I think we set a passphrase for the vault already right?

**Comment by kpcyrd** [comment]
I've added an encrypted `rebuilderd_auth_cookie` and `rebuilderd_signup_secret` secret to the repo, the vault key is expected to be located at `ansible/vault_key`. This is also documented in the README.

I've also fixed a few minor mistakes and tested the import works. There's now a backlog of packages waiting to be rebuilt:
```
core@localhost ~ $ docker exec rebuilderd rebuildctl queue ls
2021-11-06 18:26:22 libglib2.0-bin 2.70.1-1                                                                      "debian" "main" "amd64"
2021-11-06 18:26:22 libglib2.0-dev 2.70.1-1                                                                      "debian" "main" "amd64"
2021-11-06 18:26:22 dfrs 0.0.6-3                                                                                 "debian" "main" "amd64"
2021-11-06 18:26:22 sniffglue 0.14.0-1                                                                           "debian" "main" "amd64"
2021-11-06 18:26:22 anarchism 15.3-3                                                                             "debian" "main" "all"
2021-11-06 18:26:22 binutils-arm-none-eabi 15                                                                    "debian" "main" "amd64"
2021-11-06 18:26:22 librust-sniffglue-dev 0.14.0-1                                                               "debian" "main" "amd64"
core@localhost ~ $ 
```

File: ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf (hunk 1 above)
Comment (kpcyrd) [reply ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf]: Yes, I'm already using it and documented how to configure it in the readme. :)

**Comment by kpcyrd** [comment]
The change could be merged as-is imo, I can look into the status page in a follow-up PR.

**Comment by SantiagoTorres** [comment]
LGTM!

**Pull request merged**